src/events/event.ts

```diff
@@ -0,0 +1,32 @@
+export interface SerializableEvent {
+    toJSON(): Record<string, unknown>;
+}
+
+export interface DeserializableEventConstructor {
+    fromJSON(data: Record<string, unknown>): SerializableEvent;
+}
+
+export class EventSerializer {
```

Comment:
I wonder, if we're just calling `toJSON` and `fromJSON` can we jsut delete this file altogether and have the PubSubEvents call those instead?